platforms/ide/problems/src/main/java/org/gradle/problems/internal/services/ProblemsBuildTreeServices.java

```diff
@@ -74,13 +77,26 @@ ProblemSummarizer createProblemSummarizer(
         CurrentBuildOperationRef currentBuildOperationRef,
         Collection<ProblemEmitter> problemEmitters,
         InternalOptions internalOptions,
-        ProblemReportCreator problemReportCreator
+        ProblemReportCreator problemReportCreator,
+        WorkExecutionTracker workExecutionTracker
     ) {
         return new DefaultProblemSummarizer(eventEmitter,
             currentBuildOperationRef,
             ImmutableList.of(new BuildOperationBasedProblemEmitter(eventEmitter)),
             internalOptions,
-            problemReportCreator);
+            problemReportCreator,
+            id -> {
+                TaskIdentity taskIdentity = ProblemTaskIdentityTracker.getTaskIdentity();
+                if (taskIdentity != null) {
```

Comment:
I kept the threadlocal approach as an optimization. We don't need to use `AncestryTracker` for the non-worker useecase.